homeassistant/components/weatherflow/config_flow.py

```diff
@@ -0,0 +1,88 @@
+"""Config flow for WeatherFlow."""
+from __future__ import annotations
+
+import asyncio
+from asyncio import Future
+from typing import Any
+
+# import my_pypi_dependency
+from pyweatherflowudp.client import EVENT_DEVICE_DISCOVERED, WeatherFlowListener
+from pyweatherflowudp.const import DEFAULT_HOST
+from pyweatherflowudp.errors import AddressInUseError, ListenerError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.data_entry_flow import FlowResult
+
+from .const import DOMAIN, LOGGER
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST, default=DEFAULT_HOST): str})
+
+
+async def _async_has_devices(hass: HomeAssistant) -> bool:
+    """Return if there are devices that can be discovered."""
+
+    future_event: Future[bool] = asyncio.Future()
+    host: str = DEFAULT_HOST
+
+    @callback
+    def found():
```

Comment:
```suggestion
    def _async_found():
```

Reply:
does it need an `async def` as well?

Reply:
It doesn't need to be a co-routinue unless there is an `await` anything that is a callback, decorate with `@callback`

We use `async_` or `_async_` to prefix any function that needs to run in the event loop since we have sync and async code in HA

Reply:
ok good to know... thanks! (renamed)